File: 948-sort-an-array/sort-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var sortArray = function(nums) {
    if(nums.length<=1) return nums;
    let m=Math.floor(nums.length/2);
    let left=sortArray(nums.slice(0,m));
    let right=sortArray(nums.slice(m));
    return merge(left,right);
    
};
function merge(left,right){
    let result=[];
    let i=0;
    let j=0;
    while(i<left.length && j<right.length){
        if(left[i]<right[j]){
           result.push(left[i]);
           i++;
        }
        else{
            result.push(right[j]);
            j++;
        }
    }
    return [...result,...left.slice(i),...right.slice(j)];
}

File: 948-sort-an-array/sort-an-array.java
class Solution {
    public int[] sortArray(int[] nums) {
        if(nums.length<=1) return nums;
        int mid=nums.length/2;
        int[] left=sortArray(Arrays.copyOfRange(nums,0,mid));
        int[] right=sortArray(Arrays.copyOfRange(nums,mid,nums.length));
        return merge(left,right);
    }
    private int[] merge(int[] left,int[] right){
        int[] res=new int[left.length+right.length];
        int i=0,j=0,k=0;
        while(i<left.length && j<right.length){
            res[k++]=(left[i]<right[j])?left[i++]:right[j++];
        }
        while(i<left.length) res[k++]=left[i++];
        while(j<right.length) res[k++]=right[j++];
        return res;
    }
}